datasource/opennms-kafka/src/main/java/org/opennms/oce/datasource/opennms/processors/SituationTableProcessor.java

```diff
@@ -71,7 +72,28 @@ public void process(String reductionKey, OpennmsModelProtos.Alarm alarm) {
                 }
             });
         } else {
-            kvStore.delete(reductionKey);
+            OpennmsModelProtos.Alarm deletedSituation = kvStore.delete(reductionKey);
+
+            if (deletedSituation == null) {
+                LOG.warn("No situation was found in the kv store for reduction key {}", reductionKey);
+                
+                return;
+            }
+
+            situationHandlers.forEach(h -> {
+                try {
+                    //noinspection OptionalGetWithoutIsPresent, ConstantConditions
+                    String deletedSituationId = deletedSituation.getLastEvent().getParameterList().stream()
+                            .filter(parm -> parm.getName().equals(SituationToEvent.SITUATION_ID_PARM_NAME))
+                            .map(OpennmsModelProtos.EventParameter::getValue)
+                            .findFirst()
+                            .get();
```

Comment:
Can we handle failures here a little more gracefully instead of failing with a stack trace?

Reply:
What failures should I be checking? I can handle the getLastEvent == null and the parameter being empty cases with their own log to avoid catching them in the error log below, is that what you had in mind?

Reply:
Parameters being empty is the one that came to mind in which case the call to `.get()` would throw an exception.